megamek/src/megamek/client/ui/swing/util/BASE64ImageView.java

```diff
@@ -51,25 +51,31 @@ private void populateImage() {
             cache = new Hashtable<>();
             getDocument().putProperty("imageCache", cache);
         }
-
         URL src = getImageURL();
-        cache.put(src, loadImage());
-
+        Image image = loadImage();
+        if(image != null) {
+            cache.put(src, image);
+        }
     }
 
     //decodes the Base64 string into an image and returns it
     private Image loadImage() {
         String b64 = getBASE64Image();
-        BufferedImage newImage = null;
-        ByteArrayInputStream bais = null;
-        try {
-            bais = new ByteArrayInputStream(
-                    Base64.getDecoder().decode(b64.getBytes()));
-            newImage = ImageIO.read(bais);
-        } catch (Throwable ex) {
-            ex.printStackTrace();
+        if(b64 != null) {
+            BufferedImage newImage = null;
+            ByteArrayInputStream bais = null;
```

Comment:
Are we leaking these BAIS? Or do we need them held open for the BufferedImage's.

Reply:
They should be closed, I'll fix that.